src/blas/backends/portblas/portblas_common.hpp

```diff
@@ -178,6 +188,22 @@ struct throw_if_unsupported_by_device {
         throw unimplemented("blas", "portBLAS function", " for row-major");                     \
     }
 
+#define CALL_PORTBLAS_USM_FN(portblasFunc, ...)                                   \
+    if constexpr (is_column_major()) {                                            \
+        detail::throw_if_unsupported_by_device<double, sycl::aspect::fp64>{}(     \
+            " portBLAS function requiring fp64 support", __VA_ARGS__);            \
+        detail::throw_if_unsupported_by_device<sycl::half, sycl::aspect::fp16>{}( \
+            " portBLAS function requiring fp16 support", __VA_ARGS__);            \
+        auto args = detail::convert_to_portblas_type(__VA_ARGS__);                \
+        auto fn = [](auto&&... targs) {                                           \
+            return portblasFunc(std::forward<decltype(targs)>(targs)...).back();  \
+        };                                                                        \
+        return std::apply(fn, args);                                              \
+    }                                                                             \
+    else {                                                                        \
+        throw unimplemented("blas", "portBLAS function", " for row-major");       \
```

Comment:
Looks like this macro is also called for column-major, I suggest removing the row-major part of the error.